govcd/lbappprofile_test.go

```diff
@@ -0,0 +1,75 @@
+// +build lb lbAppProfile functional ALL
+
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+)
+
+func (vcd *TestVCD) Test_LBAppProfile(check *C) {
+	if vcd.config.VCD.EdgeGateway == "" {
+		check.Skip("Skipping test because no edge gateway given")
+	}
+	edge, err := vcd.vdc.FindEdgeGateway(vcd.config.VCD.EdgeGateway)
+	check.Assert(err, IsNil)
+	check.Assert(edge.EdgeGateway.Name, Equals, vcd.config.VCD.EdgeGateway)
+
+	// Used for creating
+	lbAppProfileConfig := &types.LBAppProfile{
+		Name: TestLBAppProfile,
+		Persistence: &types.LBAppProfilePersistence{
+			Method: "sourceip",
+		},
+		Template: "HTTPS",
+	}
+
+	createdLbAppProfile, err := edge.CreateLBAppProfile(lbAppProfileConfig)
+	check.Assert(err, IsNil)
+	check.Assert(createdLbAppProfile.ID, Not(IsNil))
+
+	// // We created application profile successfully therefore let's add it to cleanup list
+	parentEntity := vcd.org.Org.Name + "|" + vcd.vdc.Vdc.Name + "|" + vcd.config.VCD.EdgeGateway
+	AddToCleanupList(TestLBAppProfile, "lbAppProfile", parentEntity, check.TestName())
+
+	// // Lookup by both name and ID and compare that these are equal values
+	lbAppProfileByID, err := edge.ReadLBAppProfile(&types.LBAppProfile{ID: createdLbAppProfile.ID})
+	check.Assert(err, IsNil)
+
+	lbPoolByName, err := edge.ReadLBAppProfile(&types.LBAppProfile{Name: createdLbAppProfile.Name})
+	check.Assert(err, IsNil)
+	check.Assert(createdLbAppProfile.ID, Equals, lbPoolByName.ID)
+	check.Assert(lbAppProfileByID.ID, Equals, lbPoolByName.ID)
+	check.Assert(lbAppProfileByID.Name, Equals, lbPoolByName.Name)
+
+	check.Assert(createdLbAppProfile.Template, Equals, lbAppProfileConfig.Template)
+
+	// Test updating fields
+	// Update persistence method
+	lbAppProfileByID.Persistence.Method = "sourceip"
+	updatedAppProfile, err := edge.UpdateLBAppProfile(lbAppProfileByID)
+	check.Assert(err, IsNil)
+	check.Assert(updatedAppProfile.Persistence.Method, Equals, lbAppProfileByID.Persistence.Method)
+
+	// Verify that updated pool and its configuration are identical
+	check.Assert(updatedAppProfile, DeepEquals, lbAppProfileByID)
+
+	// Try to set invalid algorithm hash and expect API to return error
+	// Invalid persistence method invalid_method. Valid methods are: COOKIE|SSL-SESSIONID|SOURCEIP.
```

Comment:
Is API error self explanatory and we don't need validate methods types?

Reply:
It is self explanatory. In fact this line is part of API response.
`Invalid persistence method invalid_method. Valid methods are: COOKIE|SSL-SESSIONID|SOURCEIP.`